public/scripts/BulkEditOverlay.js

```diff
@@ -254,6 +260,30 @@ class BulkTagPopupHandler {
         document.querySelector('#bulk_tag_popup_reset').addEventListener('click', this.resetTags.bind(this));
         document.querySelector('#bulk_tag_popup_remove_mutual').addEventListener('click', this.removeMutual.bind(this));
         document.querySelector('#bulk_tag_popup_cancel').addEventListener('click', this.hide.bind(this));
+        document.querySelector('#bulk_tag_popup_import_all_tags').addEventListener('click', this.importAllTags.bind(this));
+        document.querySelector('#bulk_tag_popup_import_existing_tags').addEventListener('click', this.importExistingTags.bind(this));
+    }
+
+    /**
+     * Import existing tags for all selected characters
+     */
+    async importExistingTags() {
+        for (const characterId of this.characterIds) {
+            await importTags(characters[characterId], { importExisting: true });
+        }
+
+        $('#bulkTagList').empty();
+    }
```

Comment:
If you update/change the tags for characters, you need to reprint the character list in the background. That one will not refresh automatically. Check how the other functions are doing it, like reset or removing mutual.
Should simply be calling `printCharactersDebounced();`.

Goes for this and the `importAllTags` too.